src/main/asciidoc/java/http.adoc

```diff
@@ -1150,6 +1389,77 @@ httpServer.requestHandler(request -> {
 });
 ----
 
+==== Client push
+
+Server push is a new feature of HTTP/2 that enables sending multiple responses in parallel for a single client request.
+
+A push handler can be set on a request to receive the request/response pushed by the server:
+
+[source,java]
+----
+HttpClientRequest request = client.get("/index.html", response -> {
+  // Process index.html response
+});
+
+// Set a push handler to be aware of any resource pushed by the server
+request.pushHandler(pushedRequest -> {
+
+  // A resource is pushed for this request
+  System.out.println("Server pushed " + pushedRequest.path());
+
+  // Set an handler for the response
+  pushedRequest.handler(pushedResponse -> {
+    System.out.println("The response for the pushed request");
+  });
+});
+
+// End the request
+request.end();
+----
+
+If the client does not want to receive a pushed request, it can reset the stream:
+
+[source,java]
+----
+HttpClientRequest request = client.get("/index.html", response -> {
+  // Process index.html response
+});
+
+// Set a push handler to be aware of any resource pushed by the server
+request.pushHandler(pushedRequest -> {
+
+  // A resource is pushed for this request
+  System.out.println("Server pushed " + pushedRequest.path());
+
+  // Set an handler for the response
+  pushedRequest.handler(pushedResponse -> {
+    System.out.println("The response for the pushed request");
+  });
+});
+
+// End the request
+request.end();
+----
+
+When no handler is set, any stream pushed by the server will be automatically reset by the client.
```

Comment:
at the moment it sends `0` aka _ NO_ERROR_ but perhaps it should send _8_ CANCEL_